features/sessions/src/main/java/com/android254/droidconKE2020/sessions/ui/views/SessionDetailFragment.kt

```diff
@@ -2,36 +2,114 @@ package com.android254.droidconKE2020.sessions.ui.views
+import com.android254.droidconKE2020.sessions.ui.views.viewmodel.SessionDetailViewModel
+import org.koin.android.viewmodel.ext.android.viewModel
 
 
 /**
  * A simple [Fragment] subclass.
  */
 class SessionDetailFragment : Fragment(R.layout.fragment_session_detail){
+    private fun injectFeatures() = loadModules
     private var _binding : FragmentSessionDetailBinding? = null
     private val binding  get() = _binding!!
+    private val sessionDetailViewModel: SessionDetailViewModel by viewModel()
 
     override fun onCreateView(
         inflater: LayoutInflater,
         container: ViewGroup?,
         savedInstanceState: Bundle?
     ): View? {
+        injectFeatures()
         _binding = FragmentSessionDetailBinding.inflate(inflater, container, false)
+        binding.lifecycleOwner = this
+        binding.sessionDetailViewModel = sessionDetailViewModel
+        getSessionDetail()
         return binding.root
     }
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
-        binding.textViewSpeaker.setOnClickListener {
-            val speakerDetailsAction = SessionDetailFragmentDirections.actionSessionDetailsFragmentToSpeakerDetailsFragment()
-            findNavController().navigate(speakerDetailsAction)
-        }
+        observeSaveSession()
+        observeClickSpeaker()
+        observerShareSession()
+        observerNavigateBack()
+    }
+
+    private fun observerShareSession() {
+        sessionDetailViewModel.shareSession.observe(viewLifecycleOwner, Observer { session ->
+            session?.let {
+                Log.d("Fragment Tag", "We are here")
+                val shareSessionBottomFragment = ShareSessionBottomFragment(session)
+                shareSessionBottomFragment.show(parentFragmentManager, "Share Session")
+                sessionDetailViewModel.onSessionShared()
+            }
+        })
+    }
+
+    private fun observeSaveSession() {
+        sessionDetailViewModel.saveSession.observe(viewLifecycleOwner, Observer { sessionDetail ->
+            sessionDetail?.let {
+                if (sessionDetail.isSaved) {
+                    binding.imageViewSave.setImageDrawable(
+                        resources.getDrawable(
+                            R.drawable.ic_star_outline,
+                            null
+                        )
+                    )
+                    it.isSaved = false
+                    sessionDetailViewModel.onSessionSaved()
+                } else {
+                    binding.imageViewSave.setImageDrawable(
+                        resources.getDrawable(
+                            R.drawable.ic_star,
+                            null
+                        )
+                    )
+                    it.isSaved = true
+                    sessionDetailViewModel.onSessionSaved()
+                }
+            }
+        })
+    }
+
+    private fun observerNavigateBack() {
```

Comment:
What was the reason behind this, why not let Navigation Library take control of back navigation?

Reply:
Hey @wangerekaharun , actually i have no solid reason why I implemented it this way, i'll have a look at back navigation with the navigation library and update the PR. Thank you for pointing this out.

Reply:
Okay, i also go why you might have done that since we have not addressed toolbar issue and you had designed the toolbar already. There's an issue to address all the toolbars

Reply:
Once you make the commit let me know so that i can merge